satellite/metabase/rangedloop/service_test.go

```diff
@@ -43,21 +44,66 @@ func RunTest(t *testing.T, parallelism int, nSegments int, nObservers int) {
 	loopService := rangedloop.NewService(
 		zaptest.NewLogger(t),
 		rangedloop.Config{
-			BatchSize:          batchSize,
-			Parallelism:        parallelism,
-			AsOfSystemInterval: 0,
+			BatchSize:   batchSize,
+			Parallelism: parallelism,
 		},
 		&rangedlooptest.RangeSplitterMock{
 			Segments: make([]segmentloop.Segment, nSegments),
 		},
 		observers,
 	)
 
-	err := loopService.RunOnce(ctx)
+	observerDurations, err := loopService.RunOnce(ctx)
 	require.NoError(t, err)
+	require.Len(t, observerDurations, nObservers)
 
 	for _, observer := range observers {
 		countObserver := observer.(*rangedlooptest.CountObserver)
 		require.Equal(t, nSegments, countObserver.NumSegments)
 	}
 }
+
+func TestLoopDuration(t *testing.T) {
+	t.Skip("Flaky test because it validates concurrency by measuring time")
```

Comment:
In general, time.Sleep has no place in tests and rarely if ever production code for reasons like this (and others, like not being cancelable). There are strategies we can employ to have determinism with clock operations for testing that have been successful for me in the past. I won't block the review on this but thought I'd call it out.

Reply:
I agree in general but I think this is an exception because the test is verifying that execution time is measured as expected.

Reply:
Sure. I don't see one in the codebase, but this sort of thing can also be tested deterministically by introducing a clock interface that can be faked during tests. I've noticed other places in the tests that can benefit from that. I'm going to add something to storj.io/common that we can use in the future for these sorts of things.

Reply:
I don't see how its possible with a clock interface to prove that 4 batches on one worker takes 40 ms but 4 batches on 4 workers takes 10 ms.

But I'm willing to be proven wrong.

Reply:
Maybe when I'm done I'll tackle this test first :)